aidial_adapter_openai/app.py

```diff
@@ -45,9 +42,14 @@ def create_app(
     app.post("/openai/deployments/{deployment_id:path}/chat/completions")(
         endpoints.chat_completion
     )
-    app.exception_handler(OpenAIError)(openai_exception_handler)
-    app.exception_handler(pydantic.ValidationError)(pydantic_exception_handler)
-    app.exception_handler(DialException)(dial_exception_handler)
+    app.add_exception_handler(OpenAIError, openai_exception_handler)
+    app.add_exception_handler(
+        pydantic.ValidationError,
+        sdk_error_handlers.pydantic_validation_exception_handler,
+    )
+    app.add_exception_handler(
+        DialException, sdk_error_handlers.dial_exception_handler
```

Comment:
fixed - only handle openai and dial exceptions using the same exception handler. No need for the pydantic error handler since we don't validate the request using pydantic anyway.